changelog/unreleased/issue-3302

```diff
@@ -0,0 +1,8 @@
+Bugfix: Fix `fdopendir: not a directory` error for local backend
+
+The `check`, `list packs`, `prune` and `rebuild-index` commands failed
+for the local backend when the `data` folder in the repository contained
+a files. This has been fixed.
```

Comment:
Should we change "a files" to "files not created by restic"?

Reply:
Okay, reading the code it seems that you are ignoring anything that is not a directory. In that case perhaps just replace "a files" with "files" instead.

Will there be an error if there's directories in there that were not created by restic (e.g. one named `foo`)?

Reply:
@rawtaz With the un-patched `restic` 0.12.0 on macOS, adding a sub-directory in the `data` directory does not seem to have a negative impact on operation. It does not seem to matter if the directory is empty or not.

Reply:
restic traverses into all directories directly contained in `data`. As long as these directories don't contain filenames which are the hex representation of a sha256 hash, such files don't matter.